app/models/manageiq/providers/vmware/infra_manager/provision/cloning.rb

```diff
@@ -105,12 +105,16 @@ def start_clone(clone_options)
 
     [:transform, :config, :customization, :linked_clone].each { |key| vim_clone_options[key] = clone_options[key] }
 
-    [:folder, :host, :datastore, :pool, :snapshot].each do |key|
+    [:folder, :host, :pool, :snapshot].each do |key|
       ci = clone_options[key]
       next if ci.nil?
       vim_clone_options[key] = ci.ems_ref_obj
     end
 
+    vim_clone_options[:datastore] =
+      clone_options[:host].host_storages.find_by(:storage_id => clone_options[:datastore].id).ems_ref ||
+      clone_options[:datastore].ems_ref
```

Comment:
If you try to do a provision before the first ems_refresh ran it won't be set yet.  We could set it in the migration from the `Storage.ems_ref` but it'll possibly be wrong until the first refresh runs.